src/api/mvc/data-types/main.go

```diff
@@ -19,26 +19,68 @@ var variantDataTypeJson = map[string]interface{}{
 	"metadata_schema": schemas.OBJECT_SCHEMA,
 }
 
-func GetDataTypes(c echo.Context) error {
-	es := c.(*contexts.GohanContext).Es7Client
-	cfg := c.(*contexts.GohanContext).Config
+func fetchVariantData(c echo.Context) (map[string]interface{}, error) {
+	gc := c.(*contexts.GohanContext)
+	cfg := gc.Config
+	es := gc.Es7Client
 
-	// accumulate number of variants associated with each
-	// sampleId fetched from the variants overview
 	resultsMap, err := variantService.GetVariantsOverview(es, cfg)
+	if err != nil {
+		return nil, err
+	}
 
+	variantDataTypeJson["count"] = sumAllValues(resultsMap["sampleIDs"])
+	if latestCreated, ok := resultsMap["last_created_time"].(string); ok {
+		variantDataTypeJson["last_ingested"] = latestCreated
+	}
+
+	return variantDataTypeJson, nil
+}
+
+func GetDataTypes(c echo.Context) error {
+	variantData, err := fetchVariantData(c)
 	if err != nil {
-		// Could not talk to Elasticsearch, return an error
 		return c.JSON(http.StatusInternalServerError, map[string]interface{}{
 			"error": err.Error(),
 		})
 	}
 
-	variantDataTypeJson["count"] = sumAllValues(resultsMap["sampleIDs"])
+	return c.JSON(http.StatusOK, []map[string]interface{}{
+		variantData,
+	})
+}
+
+func GetReducedDataTypes(c echo.Context) error {
```

Comment:
Can you add a comment describing what this function does? How is it different from GetDataTypes?